plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/TrinoGlueCatalog.java

```diff
@@ -0,0 +1,520 @@
+    public void renameNamespace(ConnectorSession session, String source, String target)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "renameNamespace is not supported for Iceberg Glue catalogs");
+    }
+
+    @Override
+    public List<SchemaTableName> listTables(ConnectorSession session, Optional<String> namespace)
+    {
+        try {
+            return stats.getGetAllTables().call(() -> {
+                List<String> namespaces = namespace.map(List::of).orElseGet(() -> listNamespaces(session));
+                return namespaces.stream()
+                        .flatMap(glueNamespace -> {
+                            try {
+                                GetTablesRequest getTablesRequest = new GetTablesRequest().withDatabaseName(glueNamespace);
+                                catalogId.ifPresent(getTablesRequest::setCatalogId);
+                                return getPaginatedResults(
+                                        glueClient::getTables,
+                                        getTablesRequest,
+                                        GetTablesRequest::setNextToken,
+                                        GetTablesResult::getNextToken)
+                                        .map(GetTablesResult::getTableList)
+                                        .flatMap(List::stream)
+                                        .map(table -> new SchemaTableName(glueNamespace, table.getName()));
+                            }
+                            catch (EntityNotFoundException e) {
+                                return Stream.empty();
+                            }
+                        })
+                        .collect(toImmutableList());
+            });
+        }
+        catch (AmazonServiceException e) {
+            throw new TrinoException(ICEBERG_CATALOG_ERROR, e);
+        }
+    }
+
+    @Override
+    public Table loadTable(ConnectorSession session, SchemaTableName table)
+    {
+        TableMetadata metadata = tableMetadataCache.computeIfAbsent(
+                table,
+                ignore -> {
+                    TableOperations operations = tableOperationsProvider.createTableOperations(
+                            this,
+                            session,
+                            table.getSchemaName(),
+                            table.getTableName(),
+                            Optional.empty(),
+                            Optional.empty());
+                    return new BaseTable(operations, quotedTableName(table)).operations().current();
+                });
+
+        IcebergTableOperations operations = tableOperationsProvider.createTableOperations(
+                this,
+                session,
+                table.getSchemaName(),
+                table.getTableName(),
+                Optional.empty(),
+                Optional.empty());
+        operations.initializeFromMetadata(metadata);
+        return new BaseTable(operations, quotedTableName(table));
```

Comment:
would `io.trino.plugin.iceberg.IcebergUtil#getIcebergTableWithMetadata` be applicable here?

if not, we should move that method to `TrinoHiveCatalog`